src/Native/Runtime/arm/AllocFast.S

```diff
@@ -137,36 +137,36 @@ LEAF_ENTRY RhpNewArray, _TEXT
 LOCAL_LABEL(ArrayAlignSize):
         bic         r2, r2, #3
 
-        mov         r4, r0 // save EEType
-
+        push        {r0-r2}
```

Comment:
@jkotas sorry, I didn't catch you quite well, so I  just replaced stack saving of r0-r2 here by reg saving